internal/database/database_util.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"testing"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/ory/dockertest"
+
+	// imported to register the postgres migration driver
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	// imported to register the "file" source migration driver
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	// imported to register the "postgres" database driver for migrate
+)
+
+// NewTestDatabase creates a new database suitable for use in testing. This
+// should not be used outside of testing, but it is exposed in the main package
+// so it can be shared with other packages.
+//
+// All database tests can be skipped by running `go test -short` or by setting
+// the `SKIP_DATABASE_TESTS` environment variable.
+func NewTestDatabase(tb testing.TB) *DB {
+	tb.Helper()
+
+	if testing.Short() {
+		tb.Skipf("🚧 Skipping database tests (short!")
+	}
+
+	if skip, _ := strconv.ParseBool(os.Getenv("SKIP_DATABASE_TESTS")); skip {
+		tb.Skipf("🚧 Skipping database tests (SKIP_DATABASE_TESTS is set)!")
+	}
+
+	// Create the pool (docker instance).
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		tb.Fatalf("failed to create Docker pool: %s", err)
+	}
+
+	// Start the container.
+	dbname, username, password := "en-server", "my-username", "abcd1234"
```

Comment:
I'm open to generating random values here. Pretty easy to drop in `sethvargo/go-password`, but it seemed like overkill.

Reply:
I think it's Overkill since this is just for containerized tests